<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Many methods allow us to extract biological activities from omics 
    data using information from prior knowledge resources, reducing the 
    dimensionality for increased statistical power and better interpretability. 
    Here, we present decoupleR, a Bioconductor package containing different 
    statistical methods to extract these signatures within a unified framework. 
    decoupleR allows the user to flexibly test any method with any resource. 
    It incorporates methods that take into account the sign and weight of 
    network interactions. decoupleR can be used with any omic, as long as its 
    features can be linked to a biological process based on prior knowledge. 
    For example, in transcriptomics gene sets regulated by a transcription 
    factor, or in phospho-proteomics phosphosites that are targeted by a kinase.">
<title>decoupleR: Ensemble of computational methods to infer biological activities from omics data • decoupleR</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="decoupleR: Ensemble of computational methods to infer biological activities from omics data">
<meta property="og:description" content="Many methods allow us to extract biological activities from omics 
    data using information from prior knowledge resources, reducing the 
    dimensionality for increased statistical power and better interpretability. 
    Here, we present decoupleR, a Bioconductor package containing different 
    statistical methods to extract these signatures within a unified framework. 
    decoupleR allows the user to flexibly test any method with any resource. 
    It incorporates methods that take into account the sign and weight of 
    network interactions. decoupleR can be used with any omic, as long as its 
    features can be linked to a biological process based on prior knowledge. 
    For example, in transcriptomics gene sets regulated by a transcription 
    factor, or in phospho-proteomics phosphosites that are targeted by a kinase.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-11"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-11');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">decoupleR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">2.1.3</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-2">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/decoupleR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/saezlab/decoupleR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/saezlab">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="http://www.saezlab.org">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="decoupler-">decoupleR <img src="https://github.com/saezlab/decoupleR/blob/master/inst/figures/logo.svg?raw=1" align="right" width="120"><a class="anchor" aria-label="anchor" href="#decoupler-"></a>
</h1></div>
<!-- badges: start -->

<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>Many methods allow us to extract biological activities from omics data using information from prior knowledge resources, reducing the dimensionality for increased statistical power and better interpretability. Here, we present decoupleR, a Bioconductor package containing different statistical methods to extract these signatures within a unified framework. decoupleR allows the user to flexibly test any method with any resource. It incorporates methods that take into account the sign and weight of network interactions. decoupleR can be used with any omic, as long as its features can be linked to a biological process based on prior knowledge. For example, in transcriptomics gene sets regulated by a transcription factor, or in phospho-proteomics phosphosites that are targeted by a kinase.</p>
<p><img src="https://github.com/saezlab/decoupleR/blob/master/inst/figures/graphical_abstract.png?raw=1" align="center" width="800"></p>
<p>For more information about how this package has been used with real data, please check the following links:</p>
<ul>
<li><a href="https://github.com/saezlab/decoupleR_manuscript" class="external-link">decoupleR’s manuscript repository</a></li>
<li><a href="https://github.com/saezlab/decoupleRBench" class="external-link">Creation of benchmarking pipelines</a></li>
<li><a href="https://saezlab.github.io/kinase_tf_mini_tuto/" class="external-link">Example of Kinase and TF activity estimation</a></li>
</ul>
</div>
<div class="section level2">
<h2 id="installation-instructions">Installation instructions<a class="anchor" aria-label="anchor" href="#installation-instructions"></a>
</h2>
<p>Get the latest stable <code>R</code> release from <a href="http://cran.r-project.org/" class="external-link">CRAN</a>.</p>
<p>Then install <strong>decoupleR</strong> using from <a href="http://bioconductor.org/" class="external-link">Bioconductor</a> the following code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span>
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="op">}</span>

<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"decoupleR"</span><span class="op">)</span>

<span class="co"># Check that you have a valid Bioconductor installation</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/valid.html" class="external-link">valid</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p>Then install development version from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"saezlab/decoupleR"</span><span class="op">)</span></code></pre></div>
<p>Or with:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://devtools.r-lib.org/reference/remote-reexports.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"saezlab/decoupleR"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<div class="section level3">
<h3 id="load-package-and-data">Load package and data<a class="anchor" aria-label="anchor" href="#load-package-and-data"></a>
</h3>
<p>We first load the test data included inside <strong>decoupleR</strong>. It consist of a matrix (<code>mat</code>) with logFC coming from transcriptomics, and a collection of transcription factors that target gene sets with a certain mode of regulation (either positive or negative).</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/decoupleR/">decoupleR</a></span><span class="op">)</span>

<span class="va">inputs_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"testdata"</span>, <span class="st">"inputs"</span>, package <span class="op">=</span> <span class="st">"decoupleR"</span><span class="op">)</span>

<span class="va">mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">inputs_dir</span>, <span class="st">"input-expr_matrix.rds"</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt;  num [1:18490, 1:4] 3.251 0.283 -2.253 0.782 -4.575 ...</span>
<span class="co">#&gt;  - attr(*, "dimnames")=List of 2</span>
<span class="co">#&gt;   ..$ : chr [1:18490] "A1BG" "A1CF" "A2M" "A2ML1" ...</span>
<span class="co">#&gt;   ..$ : chr [1:4] "GSM2753335" "GSM2753336" "GSM2753337" "GSM2753338"</span>

<span class="va">network</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">inputs_dir</span>, <span class="st">"input-dorothea_genesets.rds"</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>
    <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>
    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://pillar.r-lib.org/reference/glimpse.html" class="external-link">glimpse</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; Rows: 151</span>
<span class="co">#&gt; Columns: 5</span>
<span class="co">#&gt; $ tf         &lt;chr&gt; "FOXO4", "FOXO4", "FOXO4", "FOXO4", "FOXO4", "FOXO4", "FOXO…</span>
<span class="co">#&gt; $ confidence &lt;chr&gt; "A", "A", "A", "A", "A", "A", "A", "A", "A", "A", "A", "A",…</span>
<span class="co">#&gt; $ target     &lt;chr&gt; "BCL2L11", "BCL6", "CDKN1A", "CDKN1B", "G6PC", "GADD45A", "…</span>
<span class="co">#&gt; $ mor        &lt;dbl&gt; 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,…</span>
<span class="co">#&gt; $ likelihood &lt;dbl&gt; 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,…</span></code></pre></div>
<p><strong>Important</strong>: Before running any method in <strong>decoupleR</strong>, we recommend the user to intersect their prior knowledge with their input matrix using <code>intersect_regulons</code>. This allows to filter out “regulons” (biological processes) with less than a minimum number of target features. We recommend to set this value (<code>minsize</code>) to at least 5:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Remove TFs with less than 5 targets in the input matrix</span>
<span class="va">network</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/intersect_regulons.html">intersect_regulons</a></span><span class="op">(</span><span class="va">mat</span>, <span class="va">network</span>, <span class="va">tf</span>, <span class="va">target</span>, minsize <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="methods">Methods<a class="anchor" aria-label="anchor" href="#methods"></a>
</h3>
<p>To check how many methods are currently available in <strong>decoupleR</strong>, run:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/show_methods.html">show_methods</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; # A tibble: 12 × 2</span>
<span class="co">#&gt;    Function      Name                                                           </span>
<span class="co">#&gt;    &lt;chr&gt;         &lt;chr&gt;                                                          </span>
<span class="co">#&gt;  1 run_aucell    AUCell                                                         </span>
<span class="co">#&gt;  2 run_consensus Consensus score between methods                                </span>
<span class="co">#&gt;  3 run_fgsea     Fast Gene Set Enrichment Analysis (FGSEA)                      </span>
<span class="co">#&gt;  4 run_gsva      Gene Set Variation Analysis (GSVA)                             </span>
<span class="co">#&gt;  5 run_mdt       Multivariate Decision Trees (MDT)                              </span>
<span class="co">#&gt;  6 run_mlm       Multivariate Linear Model (MLM)                                </span>
<span class="co">#&gt;  7 run_ora       Over Representation Analysis (ORA)                             </span>
<span class="co">#&gt;  8 run_udt       Univariate Decision Tree (UDT)                                 </span>
<span class="co">#&gt;  9 run_ulm       Univariate Linear Model (ULM)                                  </span>
<span class="co">#&gt; 10 run_viper     Virtual Inference of Protein-activity by Enriched Regulon anal…</span>
<span class="co">#&gt; 11 run_wmean     Weighted Mean (WMEAN)                                          </span>
<span class="co">#&gt; 12 run_wsum      Weighted Sum (WSUM)</span></code></pre></div>
<p>Function is the function name in <strong>decoupleR</strong> and Name is the full name of each method. To check how to use individual methods, for example <code>mlm</code>, run <code><a href="reference/run_mlm.html">?run_mlm</a></code>.</p>
<p>All methods follow the same design pattern and arguments, so moving between methods should be easy. Here is an example with <code>mlm</code>:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/run_mlm.html">run_mlm</a></span><span class="op">(</span>
    mat <span class="op">=</span> <span class="va">mat</span>,
    network <span class="op">=</span> <span class="va">network</span>,
    .source <span class="op">=</span> <span class="st">"tf"</span>,
    .target <span class="op">=</span> <span class="st">"target"</span>,
    .mor <span class="op">=</span> <span class="st">"mor"</span>,
    .likelihood <span class="op">=</span> <span class="st">"likelihood"</span>
<span class="op">)</span>
<span class="co">#&gt; # A tibble: 20 × 5</span>
<span class="co">#&gt;    statistic source condition  score p_value</span>
<span class="co">#&gt;    &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;      &lt;dbl&gt;   &lt;dbl&gt;</span>
<span class="co">#&gt;  1 mlm       FOXO4  GSM2753335 2.21  0.0288 </span>
<span class="co">#&gt;  2 mlm       NFIC   GSM2753335 1.12  0.263  </span>
<span class="co">#&gt;  3 mlm       SMAD3  GSM2753335 0.696 0.488  </span>
<span class="co">#&gt;  4 mlm       TFAP2A GSM2753335 1.34  0.183  </span>
<span class="co">#&gt;  5 mlm       RFXAP  GSM2753335 1.63  0.105  </span>
<span class="co">#&gt;  6 mlm       FOXO4  GSM2753336 2.16  0.0325 </span>
<span class="co">#&gt;  7 mlm       NFIC   GSM2753336 1.16  0.250  </span>
<span class="co">#&gt;  8 mlm       SMAD3  GSM2753336 0.882 0.380  </span>
<span class="co">#&gt;  9 mlm       TFAP2A GSM2753336 1.56  0.122  </span>
<span class="co">#&gt; 10 mlm       RFXAP  GSM2753336 2.31  0.0226 </span>
<span class="co">#&gt; 11 mlm       FOXO4  GSM2753337 2.37  0.0195 </span>
<span class="co">#&gt; 12 mlm       NFIC   GSM2753337 0.729 0.467  </span>
<span class="co">#&gt; 13 mlm       SMAD3  GSM2753337 1.11  0.270  </span>
<span class="co">#&gt; 14 mlm       TFAP2A GSM2753337 1.54  0.126  </span>
<span class="co">#&gt; 15 mlm       RFXAP  GSM2753337 2.85  0.00507</span>
<span class="co">#&gt; 16 mlm       FOXO4  GSM2753338 2.10  0.0378 </span>
<span class="co">#&gt; 17 mlm       NFIC   GSM2753338 0.550 0.584  </span>
<span class="co">#&gt; 18 mlm       SMAD3  GSM2753338 0.860 0.391  </span>
<span class="co">#&gt; 19 mlm       TFAP2A GSM2753338 1.28  0.204  </span>
<span class="co">#&gt; 20 mlm       RFXAP  GSM2753338 2.72  0.00742</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="decouple-wrapper">Decouple wrapper<a class="anchor" aria-label="anchor" href="#decouple-wrapper"></a>
</h3>
<p>Moreover, <strong>decoupleR</strong> allows to run multiple methods at the same time with the function <code><a href="reference/decouple.html">decouple()</a></code>. Statistic functions inside <strong>decoupleR</strong> always return a tidy tibble that can be easily processed with the tools provide by the <a href="https://www.tidyverse.org/" class="external-link">tidyverse ecosystem</a>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/decouple.html">decouple</a></span><span class="op">(</span>
    mat <span class="op">=</span> <span class="va">mat</span>,
    network <span class="op">=</span> <span class="va">network</span>,
    .source <span class="op">=</span> <span class="st">"tf"</span>,
    .target <span class="op">=</span> <span class="st">"target"</span>,
    statistics <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"mlm"</span>, <span class="st">"wmean"</span>, <span class="st">"ulm"</span>, <span class="st">"ora"</span><span class="op">)</span>,
    args <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>
        mlm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>center<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>,
        wmean <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>times<span class="op">=</span><span class="fl">100</span><span class="op">)</span>,
        ulm <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>center<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span>,
        ora <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>n_up<span class="op">=</span><span class="fl">150</span>, n_bottom<span class="op">=</span><span class="fl">0</span><span class="op">)</span>
    <span class="op">)</span>,
    consensus_score <span class="op">=</span> <span class="cn">TRUE</span>
<span class="op">)</span>
<span class="co">#&gt; # A tibble: 140 × 6</span>
<span class="co">#&gt;    run_id statistic source condition  score p_value</span>
<span class="co">#&gt;     &lt;dbl&gt; &lt;chr&gt;     &lt;chr&gt;  &lt;chr&gt;      &lt;dbl&gt;   &lt;dbl&gt;</span>
<span class="co">#&gt;  1      1 mlm       FOXO4  GSM2753335 2.21   0.0288</span>
<span class="co">#&gt;  2      1 mlm       NFIC   GSM2753335 1.12   0.263 </span>
<span class="co">#&gt;  3      1 mlm       SMAD3  GSM2753335 0.696  0.488 </span>
<span class="co">#&gt;  4      1 mlm       TFAP2A GSM2753335 1.34   0.183 </span>
<span class="co">#&gt;  5      1 mlm       RFXAP  GSM2753335 1.63   0.105 </span>
<span class="co">#&gt;  6      1 mlm       FOXO4  GSM2753336 2.16   0.0325</span>
<span class="co">#&gt;  7      1 mlm       NFIC   GSM2753336 1.16   0.250 </span>
<span class="co">#&gt;  8      1 mlm       SMAD3  GSM2753336 0.882  0.380 </span>
<span class="co">#&gt;  9      1 mlm       TFAP2A GSM2753336 1.56   0.122 </span>
<span class="co">#&gt; 10      1 mlm       RFXAP  GSM2753336 2.31   0.0226</span>
<span class="co">#&gt; # … with 130 more rows</span></code></pre></div>
<p>It can generate a consensus score between the methods if <code>consensus_score = TRUE</code>.</p>
</div>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>Badia-i-Mompel P., Vélez J., Braunger J., Geiss C., Dimitrov D., Müller-Dott S., Taus P., Dugourd A., Holland C.H., Ramirez Flores R.O. and Saez-Rodriguez J. 2021. decoupleR: Ensemble of computational methods to infer biological activities from omics data. bioRxiv. <a href="https://doi.org/10.1101/2021.11.04.467271" class="external-link uri">https://doi.org/10.1101/2021.11.04.467271</a></p>
</div>
<div class="section level2">
<h2 id="contributing-to-decoupler">Contributing to decoupleR<a class="anchor" aria-label="anchor" href="#contributing-to-decoupler"></a>
</h2>
<p>Are you interested in adding a new statistical method or collaborating in the development of internal tools that allow the extension of the package? Please check out our <a href="https://saezlab.github.io/decoupleR/CONTRIBUTING.html">contribution guide</a>.</p>
<hr>
<p>Please note that this project is released with a <a href="https://saezlab.github.io/decoupleR/CODE_OF_CONDUCT">Contributor Code of Conduct</a>. By participating in this project you agree to abide by its terms.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://www.bioconductor.org/packages/decoupleR" class="external-link">View on Bioconductor</a></li>
<li><a href="https://github.com/saezlab/decoupleR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/saezlab/decoupleR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a> + file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CONTRIBUTING.html">Contributing guide</a></li>
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing decoupleR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Pau Badia-i-Mompel <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-1004-3923" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Jesús Vélez <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0001-5128-3838" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Jana Braunger <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-0820-9987" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Celina Geiss <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-8740-706X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Daniel Dimitrov <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-5197-2112" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Sophia Müller-Dott <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-9710-1865" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Petr Taus <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-3764-9033" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Aurélien Dugourd <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-0714-028X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Christian H. Holland <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-3060-5786" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Ricardo O. Ramirez Flores <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-0087-371X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Julio Saez-Rodriguez <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-8552-8976" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://www.tidyverse.org/lifecycle/#maturing" class="external-link"><img src="https://img.shields.io/badge/lifecycle-maturing-blue.svg" alt="Lifecycle: maturing"></a></li>
<li><a href="https://bioconductor.org/checkResults/release/bioc-LATEST/decoupleR" class="external-link"><img src="http://www.bioconductor.org/shields/build/release/bioc/decoupleR.svg" alt="BioC status"></a></li>
<li><a href="https://bioconductor.org/checkResults/devel/bioc-LATEST/decoupleR" class="external-link"><img src="http://www.bioconductor.org/shields/build/devel/bioc/decoupleR.svg" alt="BioC dev status"></a></li>
<li><a href="https://github.com/saezlab/decoupleR/actions" class="external-link"><img src="https://github.com/saezlab/decoupleR/workflows/R-CMD-check-bioc/badge.svg" alt="R build status"></a></li>
<li><a href="https://codecov.io/gh/saezlab/decoupleR?branch=master" class="external-link"><img src="https://codecov.io/gh/saezlab/decoupleR/branch/master/graph/badge.svg" alt="Codecov test coverage"></a></li>
<li><a href="https://github.com/saezlab/decoupleR/issues" class="external-link"><img src="https://img.shields.io/github/issues/saezlab/decoupleR" alt="GitHub issues"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Pau Badia-i-Mompel, Jesús Vélez, Jana Braunger, Celina Geiss, Daniel Dimitrov, Sophia Müller-Dott, Petr Taus, Aurélien Dugourd, Christian H. Holland, Ricardo O. Ramirez Flores, Julio Saez-Rodriguez.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.1.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
