<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Changelog • decoupleR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Changelog"><meta property="og:image" content="https://saezlab.github.io/decoupleR/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-11"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-11');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">decoupleR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">2.9.7</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/decoupleR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/pw_bk.html">Pathway activity inference in bulk RNA-seq</a>
    <a class="dropdown-item" href="../articles/pw_sc.html">Pathway activity inference from scRNA-seq</a>
    <a class="dropdown-item" href="../articles/tf_bk.html">Transcription factor activity inference in bulk RNA-seq</a>
    <a class="dropdown-item" href="../articles/tf_sc.html">Transcription factor activity inference from scRNA-seq</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/saezlab/decoupleR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/saezlab">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.saezlab.org">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-news">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Changelog</h1>
      <small>Source: <a href="https://github.com/saezlab/decoupleR/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="pkg-version" data-toc-text="2.3" id="decoupler-23x">decoupleR 2.3.x<a class="anchor" aria-label="anchor" href="#decoupler-23x"></a></h2>
<div class="section level3">
<h3 id="changes-2-3">Changes<a class="anchor" aria-label="anchor" href="#changes-2-3"></a></h3>
<ul><li>
<code>ulm</code> and <code>mlm</code> are now faster but consume more memory.</li>
<li>
<code>mat</code> is now transformed to <code>matrix</code> automatically.</li>
</ul></div>
<div class="section level3">
<h3 id="new-features-2-3">New features<a class="anchor" aria-label="anchor" href="#new-features-2-3"></a></h3>
<ul><li>Added <code>get_collectri</code> wrapper to easily access the CollecTRI Gene Regulatory Network Network from <code>Omnipath</code>.</li>
<li>Added <code>get_ksn_omnipath</code> wrapper to easily access the Kinase-Substrate network from <code>Omnipath</code>.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="2.2" id="decoupler-22x">decoupleR 2.2.x<a class="anchor" aria-label="anchor" href="#decoupler-22x"></a></h2>
<div class="section level3">
<h3 id="changes-2-2">Changes<a class="anchor" aria-label="anchor" href="#changes-2-2"></a></h3>
<ul><li>Changed example <code>mat</code> and <code>net</code> to toy examples.</li>
<li>Changed test data to toy data.</li>
</ul></div>
<div class="section level3">
<h3 id="bugfixes-2-2">Bugfixes<a class="anchor" aria-label="anchor" href="#bugfixes-2-2"></a></h3>
<ul><li>
<code>ora</code> now selects correctly the top and bottom genes for p-value estimation.</li>
<li>
<code>wmean</code> and <code>wsum</code> now handle named matrices with only one sample.</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="2.1" id="decoupler-21x">decoupleR 2.1.x<a class="anchor" aria-label="anchor" href="#decoupler-21x"></a></h2>
<div class="section level3">
<h3 id="changes-2-1">Changes<a class="anchor" aria-label="anchor" href="#changes-2-1"></a></h3>
<ul><li><p><code>likelihood</code> param is deprecated, from now on, weights (positive or negative) should go to the <code>mor</code> column of <code>network</code>. Methods will still run if <code>likelihood</code> is specified, however they will be set to 1.</p></li>
<li><p>Added <code>minsize</code> argument to all methods, set to 5 by default. Sources containing less than this value of targets in the input mat will be removed from the calculations.</p></li>
<li><p>Changed default behavior of the <code>decouple</code> function. Now if no methods are specified in the <code>statistics</code> argument, the function will only run the top performers in our benchmark (<code>mlm</code>, <code>ulm</code> and <code>wsum</code>). To run all methods like before, set <code>statistics</code> to ‘all’. Moreover, the argument <code>consensus_stats</code> has been added to filter statistics for the calculation of the <code>consensus</code> score. By default it only uses <code>mlm</code>, <code>ulm</code> and <code>norm_wsum</code>, or if <code>statistics</code>==‘all’ all methods returned after running <code>decouple</code>.</p></li>
<li>
<p><code>viper</code> method:</p>
<ul><li>Now properly handles weights in <code>mor</code> by normalizing them to -1 and +1.</li>
</ul></li>
<li>
<p><code>ulm</code>/<code>mlm</code>/<code>udt</code>/<code>mdt</code> methods:</p>
<ul><li>Changed how they processed the input network. Before the model matrix only contained the intersection of features between mat and network’s targets, now it incorporates all features coming from mat ensuring a more robust prediction. Prediction values may change slightly from older versions.</li>
<li>Deprecated <code>sparse</code> argument.</li>
</ul></li>
<li>
<p><code>ora</code> method:</p>
<ul><li>Now takes top 5% features as default input instead of 300 up and bottom features.</li>
<li>Added seed to randomly break ties</li>
</ul></li>
<li>
<p><code>consensus</code> method:</p>
<ul><li>No longer based on <code>RobustRankAggreg</code>. Now the consensus score is the mean of the activities obtained after a double tailed z-score transformation.</li>
</ul></li>
<li><p>Discarded <code>filter_regulons</code> function.</p></li>
<li><p>Moved major dependencies to Suggest to reduce the number of dependencies needed.</p></li>
<li>
<p>Updated README by adding:</p>
<ul><li>Kinase inference example</li>
<li>Graphical abstract</li>
<li>Manuscript and citation</li>
<li>New vignette style</li>
</ul></li>
<li><p>Updated documentation for all methods.</p></li>
</ul></div>
<div class="section level3">
<h3 id="new-features-2-1">New features<a class="anchor" aria-label="anchor" href="#new-features-2-1"></a></h3>
<ul><li>
<p>Added wrappers to easily query <code>Omnipath</code>, one of the largest data-bases collecting prior-knowledge resources. Added these functions:</p>
<ul><li>
<code>show_resources</code>: shows available resources inside <code>Omnipath</code>.</li>
<li>
<code>get_resource</code>: gets any resource from <code>Omnipath</code>.</li>
<li>
<code>get_dorothea</code>: gets the DoRothEA gene regulatory network for transcription factor (TF) activity estimation. Note: this version is slightly different from the one in the package <code>dorothea</code> since it contains new edges and TFs and also weights the interactions by confidence levels.</li>
<li>
<code>get_progeny</code>: gets the PROGENy model for pathway activity estimation.</li>
</ul></li>
<li><p>Added <code>show_methods</code> function, it shows how many statistics are currently available.</p></li>
<li><p>Added <code>check_corr</code> function, it shows how correlated regulators in a network are. It can be used to check for co-linearity for <code>mlm</code> and <code>mdt</code>.</p></li>
<li><p>Added new error for <code>mlm</code> when co-variables are co-linear (regulators are too correlated to fit a model).</p></li>
</ul></div>
<div class="section level3">
<h3 id="bugfixes-2-1">Bugfixes<a class="anchor" aria-label="anchor" href="#bugfixes-2-1"></a></h3>
<ul><li><p><code>wmean</code> and <code>wsum</code> now return the correct empirical p-values.</p></li>
<li><p><code>ulm</code>, <code>mlm</code>, <code>mdt</code> and <code>udt</code> now accept matrices with one column as input.</p></li>
<li><p>Results from <code>ulm</code> and <code>mlm</code> now correctly return un-grouped.</p></li>
<li><p>Methods correctly run when <code>mat</code> has no column names.</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="2.0" id="decoupler-20x">decoupleR 2.0.x<a class="anchor" aria-label="anchor" href="#decoupler-20x"></a></h2>
<div class="section level3">
<h3 id="changes-2-0">Changes<a class="anchor" aria-label="anchor" href="#changes-2-0"></a></h3>
<ul><li>
<p>Some method’s names have been changed to make them easier to identify:</p>
<ul><li>
<code>pscira</code> now is called Weighted Sum (<code>wsum</code>).</li>
<li>
<code>mean</code> now is called Weighted Mean (<code>wmean</code>).</li>
<li>
<code>scira</code> now is called Univariate Linear Model (<code>ulm</code>).</li>
</ul></li>
<li><p>The column name for <code>tf</code> in the output tibbles has been changed to <code>source</code>.</p></li>
<li><p>Updated documentation for all methods.</p></li>
<li><p>Updated vignette and README.</p></li>
<li><p><code>decouple</code> function now accepts order mismatch between the list of methods and the list of methods’s arguments.</p></li>
<li><p>Moved benchmark branch to a separate repository as its own package: <a href="https://github.com/saezlab/decoupleRBench" class="external-link uri">https://github.com/saezlab/decoupleRBench</a></p></li>
</ul></div>
<div class="section level3">
<h3 id="new-features-2-0">New features<a class="anchor" aria-label="anchor" href="#new-features-2-0"></a></h3>
<ul><li>
<p>New methods added:</p>
<ul><li>Fast Gene Set Enrichment Analysis (<code>fgsea</code>).</li>
<li>
<code>AUCell</code>.</li>
<li>Univariate Decision Tree (<code>udt</code>).</li>
<li>Multivariate Decision Tree (<code>mdt</code>).</li>
<li>Multivariate Linear Model (<code>mlm</code>).</li>
</ul></li>
<li><p>New <code>decoupleR</code> manuscript repository: <a href="https://github.com/saezlab/decoupleR_manuscript" class="external-link uri">https://github.com/saezlab/decoupleR_manuscript</a></p></li>
<li><p>New <code>consensus</code> score based on <code>RobustRankAggreg::aggregateRanks()</code> added when running <code>decouple</code> with multiple methods.</p></li>
<li><p>New statistic <code>corr_wmean</code> inside <code>wmean</code>.</p></li>
<li><p>Methods based on permutations or statistical tests now return also a p-value for the obtained score (<code>fgsea</code>, <code>mlm</code>, <code>ora</code>, <code>ulm</code>, <code>viper</code>, <code>wmean</code> and <code>wsum</code>).</p></li>
<li><p>New error added when network edges are duplicated.</p></li>
<li><p>New error added when the input matrix contains NAs or Infs.</p></li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.1" id="decoupler-11x">decoupleR 1.1.x<a class="anchor" aria-label="anchor" href="#decoupler-11x"></a></h2>
<div class="section level3">
<h3 id="new-features-1-1">New features<a class="anchor" aria-label="anchor" href="#new-features-1-1"></a></h3>
<p>All new features allow for <strong>tidy selection</strong>. Making it easier to evaluate different types of data for the same method. For instance, you can specify the columns to use as strings, integer position, symbol or expression.</p>
<div class="section level4">
<h4 id="methods-1-1">Methods<a class="anchor" aria-label="anchor" href="#methods-1-1"></a></h4>
<ul><li><p>New <code><a href="../reference/decouple.html">decouple()</a></code> integrates the various member functions of the <code>decoupleR statistics</code> for centralized evaluation.</p></li>
<li>
<p>New family <code>decoupleR statists</code> for shared documentation is made up of:</p>
<ul><li>New <code><a href="../reference/run_gsva.html">run_gsva()</a></code> incorporate a convinient wrapper for <a href="https://rdrr.io/bioc/GSVA/man/gsva.html" class="external-link">GSVA::gsva()</a>.</li>
<li>New <code>run_mean()</code> calculates both the unnormalized regulatory activity and the normalized (i.e. z-score) one based on an empirical distribution.</li>
<li>New <code><a href="../reference/run_ora.html">run_ora()</a></code> fisher exact test to calculate the regulatory activity.</li>
<li>New <code>run_pscira()</code> uses a logic equivalent to <code>run_mean()</code> with the difference that it does not accept a column of likelihood.</li>
<li>New <code>run_scira()</code> calculates the regulatory activity through the coefficient <span class="math inline"><em>β</em><sub>1</sub></span> of an adjusted linear model.</li>
<li>New <code><a href="../reference/run_viper.html">run_viper()</a></code> incorporate a convinient wrapper for <a href="https://rdrr.io/bioc/viper/man/viper.html" class="external-link">viper::viper()</a>.</li>
</ul></li>
</ul></div>
<div class="section level4">
<h4 id="converters-1-1">Converters<a class="anchor" aria-label="anchor" href="#converters-1-1"></a></h4>
<ul><li>New functions family <code>convert_to_ variants</code> that allows the conversion of data to a standard format.
<ul><li>New <code>convert_to_()</code> return the entry without modification.</li>
<li>New <code>convert_to_gsva()</code> return a list of regulons suitable for <a href="https://rdrr.io/bioc/GSVA/man/gsva.html" class="external-link">GSVA::gsva()</a>.</li>
<li>New <code>convert_to_mean()</code> return a tibble with four columns: <code>tf</code>, <code>target</code>, <code>mor</code> and <code>likelihood</code>.</li>
<li>New <code>convert_to_ora()</code> returns a named list of regulons; tf with associated targets.</li>
<li>New <code>convert_to_pscira()</code> returns a tibble with three columns: <code>tf</code>, <code>target</code> and <code>mor</code>.</li>
<li>New <code>convert_to_scira()</code> returns a tibble with three columns: <code>tf</code>, <code>target</code> and <code>mor</code>.</li>
<li>New <code>convert_to_viper()</code> return a list of regulons suitable for <a href="https://rdrr.io/bioc/viper/man/viper.html" class="external-link">viper::viper()</a>
</li>
</ul></li>
</ul></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Pau Badia-i-Mompel, Jesús Vélez-Santiago, Jana Braunger, Celina Geiss, Daniel Dimitrov, Sophia Müller-Dott, Petr Taus, Aurélien Dugourd, Christian H. Holland, Ricardo O. Ramirez Flores, Julio Saez-Rodriguez.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.8.</p>
</div>

    </footer></div>

  

  

  </body></html>

