<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Calculates regulatory activities using GSVA."><title>Gene Set Variation Analysis (GSVA) — run_gsva • decoupleR</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet"><meta property="og:title" content="Gene Set Variation Analysis (GSVA) — run_gsva"><meta property="og:description" content="Calculates regulatory activities using GSVA."><meta property="og:image" content="https://saezlab.github.io/decoupleR/logo.svg"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-11"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-11');
</script></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">decoupleR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">2.9.7</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/decoupleR.html">Get started</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/pw_bk.html">Pathway activity inference in bulk RNA-seq</a>
    <a class="dropdown-item" href="../articles/pw_sc.html">Pathway activity inference from scRNA-seq</a>
    <a class="dropdown-item" href="../articles/tf_bk.html">Transcription factor activity inference in bulk RNA-seq</a>
    <a class="dropdown-item" href="../articles/tf_sc.html">Transcription factor activity inference from scRNA-seq</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/saezlab/decoupleR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/saezlab">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://www.saezlab.org">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Gene Set Variation Analysis (GSVA)</h1>
      <small class="dont-index">Source: <a href="https://github.com/saezlab/decoupleR/blob/HEAD/R/statistic-gsva.R" class="external-link"><code>R/statistic-gsva.R</code></a></small>
      <div class="d-none name"><code>run_gsva.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Calculates regulatory activities using GSVA.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">run_gsva</span><span class="op">(</span></span>
<span>  <span class="va">mat</span>,</span>
<span>  <span class="va">network</span>,</span>
<span>  .source <span class="op">=</span> <span class="va">source</span>,</span>
<span>  .target <span class="op">=</span> <span class="va">target</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gsva"</span>, <span class="st">"plage"</span>, <span class="st">"ssgsea"</span>, <span class="st">"zscore"</span><span class="op">)</span>,</span>
<span>  minsize <span class="op">=</span> <span class="fl">5L</span>,</span>
<span>  maxsize <span class="op">=</span> <span class="cn">Inf</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>mat</dt>
<dd><p>Matrix to evaluate (e.g. expression matrix).
Target nodes in rows and conditions in columns.
<code>rownames(mat)</code> must have at least one intersection with the elements
in <code>network</code> <code>.target</code> column.</p></dd>


<dt>network</dt>
<dd><p>Tibble or dataframe with edges and it's associated metadata.</p></dd>


<dt>.source</dt>
<dd><p>Column with source nodes.</p></dd>


<dt>.target</dt>
<dd><p>Column with target nodes.</p></dd>


<dt>verbose</dt>
<dd><p>Gives information about each calculation step. Default: FALSE.</p></dd>


<dt>method</dt>
<dd><p>Method to employ in the estimation of gene-set enrichment.
scores per sample. By default this is set to gsva (Hänzelmann et al, 2013).
Further available methods are "plage", "ssgsea" and "zscore". Read more in
the manual of <code><a href="https://rdrr.io/pkg/GSVA/man/gsva.html" class="external-link">GSVA::gsva</a></code>.</p></dd>


<dt>minsize</dt>
<dd><p>Integer indicating the minimum number of targets per source.
Must be greater than 0.</p></dd>


<dt>maxsize</dt>
<dd><p>Integer indicating the maximum number of targets per source.</p></dd>


<dt>...</dt>
<dd><p>Arguments passed on to <code><a href="https://rdrr.io/pkg/GSVA/man/gsvaParam-class.html" class="external-link">GSVA::gsvaParam</a></code>, <code><a href="https://rdrr.io/pkg/GSVA/man/ssgseaParam-class.html" class="external-link">GSVA::ssgseaParam</a></code></p><dl><dt><code>assay</code></dt>
<dd><p>The name of the assay to use in case <code>exprData</code> is a multi-assay
container, otherwise ignored.  By default, the first assay is used.</p></dd>

    <dt><code>annotation</code></dt>
<dd><p>The name of a Bioconductor annotation package for the gene
identifiers occurring in the row names of the expression data matrix.  This
can be used to map gene identifiers occurring in the gene sets if those are
provided in a <code>GeneSetCollection</code>.  By default gene identifiers used in
expression data matrix and gene sets are matched directly.</p></dd>

    <dt><code>kcdf</code></dt>
<dd><p>Character vector of length 1 denoting the kernel to use during
the non-parametric estimation of the cumulative distribution function of
expression levels across samples.  By default, <code>kcdf="Gaussian"</code> which is
suitable when input expression values are continuous, such as microarray
fluorescent units in logarithmic scale, RNA-seq log-CPMs, log-RPKMs or
log-TPMs.  When input expression values are integer counts, such as those
derived from RNA-seq experiments, then this argument should be set to
<code>kcdf="Poisson"</code>.</p></dd>

    <dt><code>tau</code></dt>
<dd><p>Numeric vector of length 1.  The exponent defining the weight of
the tail in the random walk performed by the <code>GSVA</code> (Hänzelmann et al.,
2013) method.  The default value is 1 as described in the paper.</p></dd>

    <dt><code>maxDiff</code></dt>
<dd><p>Logical vector of length 1 which offers two approaches to
calculate the enrichment statistic (ES) from the KS random walk statistic.</p><ul><li><p><code>FALSE</code>: ES is calculated as the maximum distance of the random walk
from 0.</p></li>
<li><p><code>TRUE</code> (the default): ES is calculated as the magnitude difference between
the largest positive and negative random walk deviations.</p></li>
</ul></dd>

    <dt><code>absRanking</code></dt>
<dd><p>Logical vector of length 1 used only when <code>maxDiff=TRUE</code>.
When <code>absRanking=FALSE</code> (default) a modified Kuiper statistic is used to
calculate enrichment scores, taking the magnitude difference between the
largest positive and negative random walk deviations. When
<code>absRanking=TRUE</code> the original Kuiper statistic that sums the largest
positive and negative random walk deviations, is used. In this latter case,
gene sets with genes enriched on either extreme (high or low) will be
regarded as ’highly’ activated.</p></dd>

    <dt><code>alpha</code></dt>
<dd><p>Numeric vector of length 1.  The exponent defining the
weight of the tail in the random walk performed by the <code>ssGSEA</code> (Barbie et
al., 2009) method.  The default value is 0.25 as described in the paper.</p></dd>

    <dt><code>normalize</code></dt>
<dd><p>Logical vector of length 1; if <code>TRUE</code>  runs the <code>ssGSEA</code> method
from Barbie et al. (2009) normalizing the scores by the absolute difference
between the minimum and the maximum, as described in their paper. Otherwise
this last normalization step is skipped.</p></dd>

  
</dl></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>A long format tibble of the enrichment scores for each source
across the samples. Resulting tibble contains the following columns:</p><ol><li><p><code>statistic</code>: Indicates which method is associated with which score.</p></li>
<li><p><code>source</code>: Source nodes of <code>network</code>.</p></li>
<li><p><code>condition</code>: Condition representing each column of <code>mat</code>.</p></li>
<li><p><code>score</code>: Regulatory activity (enrichment score).</p></li>
</ol></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>GSVA (Hänzelmann et al., 2013) starts by transforming the input molecular
readouts in mat to a readout-level statistic using Gaussian kernel estimation
of the cumulative density function. Then, readout-level statistics are
ranked per sample and normalized to up-weight the two tails of the rank
distribution. Afterwards, an enrichment score <code>gsva</code> is calculated
using a running sum statistic that is normalized by subtracting the largest
negative estimate from the largest positive one.</p>
<p>Hänzelmann S. et al. (2013) GSVA: gene set variation analysis for microarray
and RNA-seq data. BMC Bioinformatics, 14, 7.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p>Other decoupleR statistics: 
<code><a href="decouple.html">decouple</a>()</code>,
<code><a href="run_aucell.html">run_aucell</a>()</code>,
<code><a href="run_fgsea.html">run_fgsea</a>()</code>,
<code><a href="run_mdt.html">run_mdt</a>()</code>,
<code><a href="run_mlm.html">run_mlm</a>()</code>,
<code><a href="run_ora.html">run_ora</a>()</code>,
<code><a href="run_udt.html">run_udt</a>()</code>,
<code><a href="run_ulm.html">run_ulm</a>()</code>,
<code><a href="run_viper.html">run_viper</a>()</code>,
<code><a href="run_wmean.html">run_wmean</a>()</code>,
<code><a href="run_wsum.html">run_wsum</a>()</code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">inputs_dir</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"testdata"</span>, <span class="st">"inputs"</span>, package <span class="op">=</span> <span class="st">"decoupleR"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">inputs_dir</span>, <span class="st">"mat.rds"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">net</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">inputs_dir</span>, <span class="st">"net.rds"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">run_gsva</span><span class="op">(</span><span class="va">mat</span>, <span class="va">net</span>, minsize<span class="op">=</span><span class="fl">1</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 72 × 4</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    statistic source condition score</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 1</span> gsva      T1     S01       0.222</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 2</span> gsva      T1     S02       0.556</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 3</span> gsva      T1     S03       0.667</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 4</span> gsva      T1     S04       0.778</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 5</span> gsva      T1     S05       0.556</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 6</span> gsva      T1     S06       0.667</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 7</span> gsva      T1     S07       0.667</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 8</span> gsva      T1     S08       0.667</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;"> 9</span> gsva      T1     S09       0.889</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">10</span> gsva      T1     S10       0.444</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># ℹ 62 more rows</span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Pau Badia-i-Mompel, Jesús Vélez-Santiago, Jana Braunger, Celina Geiss, Daniel Dimitrov, Sophia Müller-Dott, Petr Taus, Aurélien Dugourd, Christian H. Holland, Ricardo O. Ramirez Flores, Julio Saez-Rodriguez.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.8.</p>
</div>

    </footer></div>

  

  

  </body></html>

