<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="decoupleR">
<title>Introduction • decoupleR</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction">
<meta property="og:description" content="decoupleR">
<meta property="og:image" content="https://saezlab.github.io/decoupleR/logo.svg">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-119440867-11"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-119440867-11');
</script>
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">decoupleR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">2.1.9</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item">
  <a class="nav-link" href="../articles/decoupleR.html">Get started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/pw_bk.html">Pathway activity inference in bulk RNA-seq</a>
    <a class="dropdown-item" href="../articles/pw_sc.html">Pathway activity inference from scRNA-seq</a>
    <a class="dropdown-item" href="../articles/tf_bk.html">Transcription factor activity inference in bulk RNA-seq</a>
    <a class="dropdown-item" href="../articles/tf_sc.html">Transcription factor activity inference from scRNA-seq</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/saezlab/decoupleR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="https://twitter.com/saezlab">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li class="nav-item">
  <a class="external-link nav-link" href="http://www.saezlab.org">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Introduction</h1>
                        <h4 data-toc-skip class="author">Pau
Badia-i-Mompel</h4>
            <address class="author_afil">
      Heidelberg
Universiy<br><h4 data-toc-skip class="author">Jesús
Vélez-Santiago</h4>
            <address class="author_afil">
      National Autonomous University of Mexico<br><small class="dont-index">Source: <a href="https://github.com/saezlab/decoupleR/blob/HEAD/vignettes/decoupleR.Rmd" class="external-link"><code>vignettes/decoupleR.Rmd</code></a></small>
      <div class="d-none name"><code>decoupleR.Rmd</code></div>
    </address>
</address>
</div>

    
    
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.14/decoupleR" class="external-link">decoupleR</a></em>
is an R package distributed as part of the Bioconductor project. To
install the package, start R and enter:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span>
<span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"decoupleR"</span><span class="op">)</span></code></pre></div>
<p>Alternatively, you can instead install the latest development version
from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"saezlab/decoupleR"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p><em><a href="https://bioconductor.org/packages/3.14/decoupleR" class="external-link">decoupleR</a></em>
(Badia-i-Mompel, Santiago, Braunger, Geiss, Dimitrov, Müller-Dott, Taus,
Dugourd, Holland, Flores, and Saez-Rodriguez, 2022) contains different
statistical methods to extract biological activities from omics data
using prior knowledge. Some of them are:</p>
<ul>
<li>AUCell: (Aibar, Bravo Gonzalez-Blas, Moerman, Huynh-Thu, Imrichova,
Hulselmans, Rambow, Marine, Geurts, Aerts, van den Oord, Kalender Atak,
Wouters, and Aerts, 2017)</li>
<li>Fast GSEA: (Korotkevich, Sukhov, and Sergushichev, 2019)</li>
<li>GSVA: (H{ä}nzelmann, Castelo, and Guinney, 2013)</li>
<li>viper: (Alvarez, Shen, Giorgi, Lachmann, Ding, Ye, and Califano,
2016)</li>
</ul>
<p>In this vignette we showcase how to use it with some toy data.</p>
<div class="section level3">
<h3 id="libraries">Libraries<a class="anchor" aria-label="anchor" href="#libraries"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.14/decoupleR" class="external-link">decoupleR</a></em>
can be imported as:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://saezlab.github.io/decoupleR/">decoupleR</a></span><span class="op">)</span>

<span class="co"># Extra libraries</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="input-data">Input data<a class="anchor" aria-label="anchor" href="#input-data"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.14/decoupleR" class="external-link">decoupleR</a></em>
needs a matrix (<code>mat</code>) of any molecular readouts (gene
expression, logFC, p-values, etc.) and a <code>network</code> that
relates target features (genes, proteins, etc.) to “source” biological
entities (pathways, transcription factors, molecular processes, etc.).
Some methods also require the mode of regulation (MoR) for each
interaction, defined as negative or positive weights.</p>
<p>To get an example data-set, run:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_toy_data.html">get_toy_data</a></span><span class="op">(</span><span class="op">)</span>

<span class="va">mat</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">mat</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mat</span>,<span class="fl">5</span><span class="op">)</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>
<span class="co">#&gt;           S01       S02       S03        S04      S05</span>
<span class="co">#&gt; G01 9.3709584 9.3888607 9.8951935  8.7844590 8.431446</span>
<span class="co">#&gt; G02 8.5646982 8.2787888 8.4304691  8.8509076 8.655648</span>
<span class="co">#&gt; G03 8.3631284 8.1333213 8.2572694 10.4142076 8.321925</span>
<span class="co">#&gt; G04 8.6328626 8.6359504 9.7631631  8.0361226 8.783839</span>
<span class="co">#&gt; G05 0.4042683 0.2842529 0.4600974  0.2059986 1.575728</span>

<span class="va">network</span> <span class="op">&lt;-</span> <span class="va">data</span><span class="op">$</span><span class="va">network</span>
<span class="va">network</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 3</span></span>
<span class="co">#&gt;    source target   mor</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> T1     G01      1  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> T1     G02      1  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> T1     G03      0.7</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> T2     G06      1  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> T2     G07      0.5</span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> T2     G08      1  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> T3     G06     -<span style="color: #BB0000;">0.5</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> T3     G07     -<span style="color: #BB0000;">3</span>  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> T3     G08     -<span style="color: #BB0000;">1</span>  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> T3     G11      1</span></code></pre></div>
<p>This example consists of two small populations of samples (S, cols)
with different gene expression patterns (G, rows):</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">mat</span>, cluster_rows <span class="op">=</span> <span class="cn">F</span>, cluster_cols <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></code></pre></div>
<p><img src="decoupleR_files/figure-html/show_matrix-1.png" width="700"></p>
<p>Here we can see that some genes seem to be more expressed in one
group of samples than in the other and vice-versa. Ideally, we would
like to capture these differences in gene programs into interpretable
biological entities. In this example we will do it by summarizing gene
expression into transcription factor activities.</p>
<p>The toy data also contains a simple net consisting of 3 transcription
factors (Ts) with specific regulation to target genes (either positive
or negative). This network can be visualized like a graph. Green edges
are positive regulation (activation), red edges are negative regulation
(inactivation):</p>
<p><img src="https://github.com/saezlab/decoupleR/blob/master/inst/figures/net_plot.png?raw=1" align="center" width="600"></p>
<p>According to this network, the first population of samples should
show high activity for T1 and T3, while the second one only for T2.</p>
</div>
<div class="section level3">
<h3 id="methods">Methods<a class="anchor" aria-label="anchor" href="#methods"></a>
</h3>
<p><em><a href="https://bioconductor.org/packages/3.14/decoupleR" class="external-link">decoupleR</a></em>
contains several methods. To check how many are available, run:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/show_methods.html">show_methods</a></span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 12 × 2</span></span>
<span class="co">#&gt;    Function      Name                                                           </span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>         <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                                          </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> run_aucell    AUCell                                                         </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> run_consensus Consensus score between methods                                </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> run_fgsea     Fast Gene Set Enrichment Analysis (FGSEA)                      </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> run_gsva      Gene Set Variation Analysis (GSVA)                             </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> run_mdt       Multivariate Decision Trees (MDT)                              </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> run_mlm       Multivariate Linear Model (MLM)                                </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> run_ora       Over Representation Analysis (ORA)                             </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> run_udt       Univariate Decision Tree (UDT)                                 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> run_ulm       Univariate Linear Model (ULM)                                  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> run_viper     Virtual Inference of Protein-activity by Enriched Regulon anal…</span>
<span class="co">#&gt; <span style="color: #BCBCBC;">11</span> run_wmean     Weighted Mean (WMEAN)                                          </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">12</span> run_wsum      Weighted Sum (WSUM)</span></code></pre></div>
<p>Each method models biological activities in a different manner,
sometimes returning more than one estimate or providing significance of
the estimation. To know what each method returns, please check their
documentation like this <code><a href="../reference/run_mlm.html">?run_mlm</a></code>.</p>
<p>To have a unified framework, methods have these shared arguments:</p>
<ul>
<li>
<code>mat</code> : input matrix of molecular readouts.</li>
<li>
<code>network</code> : input prior knowledge information relating
molecular features to biological entities.</li>
<li>
<code>.source</code>,<code>.target</code> and <code>.mor</code> :
column names where to extract the information from <code>network</code>.
<ul>
<li>
<code>.source</code> refers to the biological entities.</li>
<li>
<code>.target</code> refers to the molecular features.</li>
<li>
<code>.mor</code> refers to the “strength” of the interaction (if
available, else 1s will be used). Only available for methods that can
model interaction weights.</li>
</ul>
</li>
<li>
<code>minsize</code> : Minimum of target features per biological
entity (5 by default). If less, sources are removed. This filtering
prevents obtaining noisy activities from biological entities with very
few matching target features in <code>matrix</code>. For this example
data-set we will have to keep it to 0 though.</li>
</ul>
</div>
<div class="section level3">
<h3 id="running-methods">Running methods<a class="anchor" aria-label="anchor" href="#running-methods"></a>
</h3>
<div class="section level4">
<h4 id="individual-methods">Individual methods<a class="anchor" aria-label="anchor" href="#individual-methods"></a>
</h4>
<p>As an example, let’s first run the Gene Set Enrichment Analysis
method (<code>gsea</code>), one of the most well-known statistics:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_gsea</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_fgsea.html">run_fgsea</a></span><span class="op">(</span><span class="va">mat</span>, <span class="va">network</span>, .source<span class="op">=</span><span class="st">'source'</span>, .target<span class="op">=</span><span class="st">'target'</span>, nproc<span class="op">=</span><span class="fl">1</span>, minsize <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>
<span class="va">res_gsea</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 144 × 5</span></span>
<span class="co">#&gt;    statistic  source condition   score p_value</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> fgsea      T1     S01         0.889  0.075<span style="text-decoration: underline;">5</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> norm_fgsea T1     S01         1.24   0.075<span style="text-decoration: underline;">5</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> fgsea      T2     S01        -<span style="color: #BB0000;">0.667</span>  0.548 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> norm_fgsea T2     S01        -<span style="color: #BB0000;">1.11</span>   0.548 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> fgsea      T3     S01        -<span style="color: #BB0000;">0.75</span>   0.01  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> norm_fgsea T3     S01       <span style="color: #BB0000;">Inf</span>      0.01  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> fgsea      T1     S02         0.889  0.076<span style="text-decoration: underline;">4</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> norm_fgsea T1     S02         1.29   0.076<span style="text-decoration: underline;">4</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> fgsea      T2     S02         0      0.977 </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> norm_fgsea T2     S02         0      0.977 </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 134 more rows</span></span></code></pre></div>
<p>Methods return a result data-frame containing:</p>
<ul>
<li>
<code>statistic</code>: name of the statistic. Depending on the
method, there can be more than one per method.</li>
<li>
<code>source</code>: name of the biological entity.</li>
<li>
<code>condition</code>: sample name.</li>
<li>
<code>score</code>: inferred biological activity.</li>
<li>
<code>p_value</code>: if available, significance of the inferred
activity.</li>
</ul>
<p>In the case of <code>gsea</code>, it returns a simple estimate of
activities (<code>fgsea</code>), a normalized estimate
(<code>norm_fgsea</code>) and p-values after doing permutations.</p>
<p>Other methods can return different things, for example Univariate
Linear Model (<code>ulm</code>):</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_ulm</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_ulm.html">run_ulm</a></span><span class="op">(</span><span class="va">mat</span>, <span class="va">network</span>, .source<span class="op">=</span><span class="st">'source'</span>, .target<span class="op">=</span><span class="st">'target'</span>, .mor<span class="op">=</span><span class="st">'mor'</span>, minsize <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>
<span class="va">res_ulm</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 72 × 5</span></span>
<span class="co">#&gt;    statistic source condition score  p_value</span>
<span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> ulm       T1     S01        4.21 0.001<span style="text-decoration: underline;">80</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> ulm       T1     S02        4.07 0.002<span style="text-decoration: underline;">24</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> ulm       T1     S03        3.85 0.003<span style="text-decoration: underline;">19</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> ulm       T1     S04        4.60 0.000<span style="text-decoration: underline;">979</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> ulm       T1     S05        3.90 0.002<span style="text-decoration: underline;">98</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> ulm       T1     S06        3.66 0.004<span style="text-decoration: underline;">42</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> ulm       T1     S07        4.31 0.001<span style="text-decoration: underline;">53</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> ulm       T1     S08        4.65 0.000<span style="text-decoration: underline;">902</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> ulm       T1     S09        4.49 0.001<span style="text-decoration: underline;">17</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span> ulm       T1     S10        4.07 0.002<span style="text-decoration: underline;">25</span> </span>
<span class="co">#&gt; <span style="color: #949494;"># … with 62 more rows</span></span></code></pre></div>
<p>In this case, <code>ulm</code> returns just an estimate
(<code>ulm</code>) and its associated p-values. Each method can return
different statistics, we recommend to check their documentation to know
more about them.</p>
<p>Let us plot the obtained results, first for <code>gsea</code>:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Transform to matrix</span>
<span class="va">mat_gsea</span> <span class="op">&lt;-</span> <span class="va">res_gsea</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">statistic</span><span class="op">==</span><span class="st">'fgsea'</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="../reference/pivot_wider_profile.html">pivot_wider_profile</a></span><span class="op">(</span>id_cols <span class="op">=</span> <span class="va">source</span>, names_from <span class="op">=</span> <span class="va">condition</span>, 
                      values_from <span class="op">=</span> <span class="va">score</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">mat_gsea</span>, cluster_rows <span class="op">=</span> <span class="cn">F</span>, cluster_cols <span class="op">=</span> <span class="cn">F</span>, cellwidth <span class="op">=</span> <span class="fl">15</span>, cellheight <span class="op">=</span> <span class="fl">40</span><span class="op">)</span></code></pre></div>
<p><img src="decoupleR_files/figure-html/res_gsea-1.png" width="700"></p>
<p>We can observe that for transcription factors T1 and T2, the obtained
activities correctly distinguish the two sample populations. T3, on the
other hand, should be down for the second population of samples since it
is a repressor. This mislabeling of activities happens because
<code>gsea</code> cannot model weights when inferring biological
activities.</p>
<p>When weights are available in the prior knowledge, we definitely
recommend using any of the methods that take them into account to get
better estimates, one example is <code>ulm</code>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Transform to matrix</span>
<span class="va">mat_ulm</span> <span class="op">&lt;-</span> <span class="va">res_ulm</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">statistic</span><span class="op">==</span><span class="st">'ulm'</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="../reference/pivot_wider_profile.html">pivot_wider_profile</a></span><span class="op">(</span>id_cols <span class="op">=</span> <span class="va">source</span>, names_from <span class="op">=</span> <span class="va">condition</span>, 
                      values_from <span class="op">=</span> <span class="va">score</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">mat_ulm</span>, cluster_rows <span class="op">=</span> <span class="cn">F</span>, cluster_cols <span class="op">=</span> <span class="cn">F</span>, cellwidth <span class="op">=</span> <span class="fl">15</span>, cellheight <span class="op">=</span> <span class="fl">40</span><span class="op">)</span></code></pre></div>
<p><img src="decoupleR_files/figure-html/res_ulm-1.png" width="700"></p>
<p>Since <code>ulm</code> models weights when estimating biological
activities, it correctly assigns T3 as inactive in the second population
of samples.</p>
</div>
<div class="section level4">
<h4 id="multiple-methods">Multiple methods<a class="anchor" aria-label="anchor" href="#multiple-methods"></a>
</h4>
<p><em><a href="https://bioconductor.org/packages/3.14/decoupleR" class="external-link">decoupleR</a></em>
also allows to run multiple methods at the same time. Moreover, it
computes a consensus score based on the obtained activities across
methods, called <code>consensus</code>.</p>
<p>By default, <code>deocuple</code> runs only the top performer methods
in our benchmark (<code>mlm</code>, <code>ulm</code> and
<code>wsum</code>), and estimates a consensus score across them.
Specific arguments to specific methods can be passed using the variable
<code>args</code>. For more information check
<code><a href="../reference/decouple.html">?decouple</a></code>.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">res_decouple</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/decouple.html">decouple</a></span><span class="op">(</span><span class="va">mat</span>, 
                         <span class="va">network</span>, 
                         .source<span class="op">=</span><span class="st">'source'</span>, 
                         .target<span class="op">=</span><span class="st">'target'</span>,
                         minsize <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>
<span class="va">res_decouple</span>
<span class="co">#&gt; <span style="color: #949494;"># A tibble: 432 × 6</span></span>
<span class="co">#&gt;    run_id statistic source condition   score p_value</span>
<span class="co">#&gt;     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span>      1 mlm       T1     S01        3.52   0.007<span style="text-decoration: underline;">81</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span>      1 mlm       T2     S01       -<span style="color: #BB0000;">1.13</span>   0.290  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span>      1 mlm       T3     S01       -<span style="color: #BB0000;">0.247</span>  0.811  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span>      1 mlm       T1     S02        3.48   0.008<span style="text-decoration: underline;">31</span></span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span>      1 mlm       T2     S02       -<span style="color: #BB0000;">0.213</span>  0.837  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span>      1 mlm       T3     S02       -<span style="color: #BB0000;">0.353</span>  0.733  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span>      1 mlm       T1     S03        3.15   0.013<span style="text-decoration: underline;">5</span> </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span>      1 mlm       T2     S03       -<span style="color: #BB0000;">0.638</span>  0.541  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span>      1 mlm       T3     S03        0.074<span style="text-decoration: underline;">9</span> 0.942  </span>
<span class="co">#&gt; <span style="color: #BCBCBC;">10</span>      1 mlm       T1     S04        3.82   0.005<span style="text-decoration: underline;">12</span></span>
<span class="co">#&gt; <span style="color: #949494;"># … with 422 more rows</span></span></code></pre></div>
<p>Let us see the result for the consensus score in the previous
<code>decouple</code> run:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># Transform to matrix</span>
<span class="va">mat_consensus</span> <span class="op">&lt;-</span> <span class="va">res_decouple</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">statistic</span><span class="op">==</span><span class="st">'consensus'</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="../reference/pivot_wider_profile.html">pivot_wider_profile</a></span><span class="op">(</span>id_cols <span class="op">=</span> <span class="va">source</span>, names_from <span class="op">=</span> <span class="va">condition</span>, 
                      values_from <span class="op">=</span> <span class="va">score</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">as.matrix</a></span><span class="op">(</span><span class="op">)</span>

<span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">mat_consensus</span>, cluster_rows <span class="op">=</span> <span class="cn">F</span>, cluster_cols <span class="op">=</span> <span class="cn">F</span>, cellwidth <span class="op">=</span> <span class="fl">15</span>, cellheight <span class="op">=</span> <span class="fl">40</span><span class="op">)</span></code></pre></div>
<p><img src="decoupleR_files/figure-html/res_decouple-1.png" width="700"></p>
<p>We can observe that the consensus score correctly predicts that T1
and T3 should be active in the first population of samples while T2 in
the second one.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="session-information">Session information<a class="anchor" aria-label="anchor" href="#session-information"></a>
</h2>
<pre><code><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span>
<span class="co">#&gt;  version  R version 4.1.3 (2022-03-10)</span>
<span class="co">#&gt;  os       Ubuntu 20.04.4 LTS</span>
<span class="co">#&gt;  system   x86_64, linux-gnu</span>
<span class="co">#&gt;  ui       X11</span>
<span class="co">#&gt;  language en</span>
<span class="co">#&gt;  collate  en_US.UTF-8</span>
<span class="co">#&gt;  ctype    en_US.UTF-8</span>
<span class="co">#&gt;  tz       UTC</span>
<span class="co">#&gt;  date     2022-04-11</span>
<span class="co">#&gt;  pandoc   2.17.1.1 @ /usr/local/bin/ (via rmarkdown)</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span>
<span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package     </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version</span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span>
<span class="co">#&gt;  assertthat     0.2.1   <span style="color: #555555;">2019-03-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  BiocManager    1.30.16 <span style="color: #555555;">2021-06-15</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.1.3)</span></span>
<span class="co">#&gt;  BiocParallel   1.28.3  <span style="color: #555555;">2021-12-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  BiocStyle    * 2.22.0  <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  bookdown       0.25    <span style="color: #555555;">2022-03-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  bslib          0.3.1   <span style="color: #555555;">2021-10-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  cachem         1.0.6   <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  cli            3.2.0   <span style="color: #555555;">2022-02-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  colorspace     2.0-3   <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  crayon         1.5.1   <span style="color: #555555;">2022-03-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  data.table     1.14.2  <span style="color: #555555;">2021-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  DBI            1.1.2   <span style="color: #555555;">2021-12-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  decoupleR    * 2.1.9   <span style="color: #555555;">2022-04-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  desc           1.4.1   <span style="color: #555555;">2022-03-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  digest         0.6.29  <span style="color: #555555;">2021-12-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  dplyr        * 1.0.8   <span style="color: #555555;">2022-02-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  ellipsis       0.3.2   <span style="color: #555555;">2021-04-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  evaluate       0.15    <span style="color: #555555;">2022-02-18</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  fansi          1.0.3   <span style="color: #555555;">2022-03-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  fastmap        1.1.0   <span style="color: #555555;">2021-01-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  fastmatch      1.1-3   <span style="color: #555555;">2021-07-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  fgsea          1.20.0  <span style="color: #555555;">2021-10-26</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span>
<span class="co">#&gt;  fs             1.5.2   <span style="color: #555555;">2021-12-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  generics       0.1.2   <span style="color: #555555;">2022-01-31</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  ggplot2        3.3.5   <span style="color: #555555;">2021-06-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  glue           1.6.2   <span style="color: #555555;">2022-02-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  gridExtra      2.3     <span style="color: #555555;">2017-09-09</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  gtable         0.3.0   <span style="color: #555555;">2019-03-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  highr          0.9     <span style="color: #555555;">2021-04-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  htmltools      0.5.2   <span style="color: #555555;">2021-08-25</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  httr           1.4.2   <span style="color: #555555;">2020-07-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  jquerylib      0.1.4   <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  jsonlite       1.8.0   <span style="color: #555555;">2022-02-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  knitr          1.38    <span style="color: #555555;">2022-03-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  lattice        0.20-45 <span style="color: #555555;">2021-09-22</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.1.3)</span></span>
<span class="co">#&gt;  lifecycle      1.0.1   <span style="color: #555555;">2021-09-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  lubridate      1.8.0   <span style="color: #555555;">2021-10-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  magrittr       2.0.3   <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  Matrix         1.4-1   <span style="color: #555555;">2022-03-23</span> <span style="color: #555555;">[3]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  memoise        2.0.1   <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  munsell        0.5.0   <span style="color: #555555;">2018-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  pheatmap     * 1.0.12  <span style="color: #555555;">2019-01-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  pillar         1.7.0   <span style="color: #555555;">2022-02-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  pkgconfig      2.0.3   <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  pkgdown        2.0.2   <span style="color: #555555;">2022-01-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  plyr           1.8.7   <span style="color: #555555;">2022-03-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  purrr          0.3.4   <span style="color: #555555;">2020-04-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  R6             2.5.1   <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  ragg           1.2.2   <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  RColorBrewer   1.1-3   <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  Rcpp           1.0.8.3 <span style="color: #555555;">2022-03-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  RefManageR   * 1.3.0   <span style="color: #555555;">2020-11-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  rlang          1.0.2   <span style="color: #555555;">2022-03-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  rmarkdown      2.13    <span style="color: #555555;">2022-03-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  rprojroot      2.0.3   <span style="color: #555555;">2022-04-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  sass           0.4.1   <span style="color: #555555;">2022-03-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  scales         1.1.1   <span style="color: #555555;">2020-05-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  sessioninfo    1.2.2   <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  stringi        1.7.6   <span style="color: #555555;">2021-11-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  stringr        1.4.0   <span style="color: #555555;">2019-02-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  systemfonts    1.0.4   <span style="color: #555555;">2022-02-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  textshaping    0.3.6   <span style="color: #555555;">2021-10-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  tibble         3.1.6   <span style="color: #555555;">2021-11-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  tidyr          1.2.0   <span style="color: #555555;">2022-02-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  tidyselect     1.1.2   <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  utf8           1.2.2   <span style="color: #555555;">2021-07-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  vctrs          0.4.0   <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  withr          2.5.0   <span style="color: #555555;">2022-03-03</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  xfun           0.30    <span style="color: #555555;">2022-03-02</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  xml2           1.3.3   <span style="color: #555555;">2021-11-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt;  yaml           2.3.5   <span style="color: #555555;">2022-02-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.1.0)</span></span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span>
<span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span>
<span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span>
<span class="co">#&gt; </span>
<span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>
<a id="bib-aibar2017scenic"></a><a href="#cite-aibar2017scenic">[1]</a><cite>
S. Aibar, C. Bravo Gonzalez-Blas, T. Moerman, et al. “SCENIC:
Single-Cell Regulatory Network Inference And Clustering”. In: <em>Nature
Methods</em> 14 (2017), pp. 1083-1086. DOI:
<a href="https://doi.org/10.1038/nmeth.4463" class="external-link">10.1038/nmeth.4463</a>.</cite>
</p>
<p>
<a id="bib-alvarez2016functional"></a><a href="#cite-alvarez2016functional">[2]</a><cite>
M. J. Alvarez, Y. Shen, F. M. Giorgi, et al. “Functional
characterization of somatic mutations in cancer using network-based
inference of protein activity”. In: <em>Nature genetics</em> 48.8
(2016), pp. 838–47.</cite>
</p>
<p>
<a id="bib-badiaimompel2022decoupler"></a><a href="#cite-badiaimompel2022decoupler">[3]</a><cite>
P. Badia-i-Mompel, J. V. Santiago, J. Braunger, et al. “decoupleR:
ensemble of computational methods to infer biological activities from
omics data”. In: <em>Bioinformatics Advances</em> (2022). DOI:
<a href="https://doi.org/https://doi.org/10.1093/bioadv/vbac016" class="external-link">https://doi.org/10.1093/bioadv/vbac016</a>.</cite>
</p>
<p>
<a id="bib-hanzelmann2013gsva"></a><a href="#cite-hanzelmann2013gsva">[4]</a><cite>
S. Hänzelmann, R. Castelo, and J. Guinney. “GSVA: gene set variation
analysis for microarray and RNA-Seq data”. In: <em>BMC
Bioinformatics</em> 14 (2013), p. 7. DOI:
<a href="https://doi.org/10.1186/1471-2105-14-7" class="external-link">10.1186/1471-2105-14-7</a>.
URL:
<a href="http://www.biomedcentral.com/1471-2105/14/7" class="external-link">http://www.biomedcentral.com/1471-2105/14/7</a>.</cite>
</p>
<p>
<a id="bib-korotkevich2019fast"></a><a href="#cite-korotkevich2019fast">[5]</a><cite>
G. Korotkevich, V. Sukhov, and A. Sergushichev. “Fast gene set
enrichment analysis”. In: <em>bioRxiv</em> (2019). DOI:
<a href="https://doi.org/10.1101/060012" class="external-link">10.1101/060012</a>. URL:
<a href="http://biorxiv.org/content/early/2016/06/20/060012" class="external-link">http://biorxiv.org/content/early/2016/06/20/060012</a>.</cite>
</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Pau Badia-i-Mompel, Jesús Vélez-Santiago, Jana Braunger,
Celina Geiss, Daniel Dimitrov, Sophia Müller-Dott, Petr Taus, Aurélien
Dugourd, Christian H. Holland, Ricardo O. Ramirez Flores, Julio
Saez-Rodriguez.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a>
2.0.2.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
